ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs

```diff
@@ -482,6 +488,48 @@ where
     Ok(changed_keys)
 }
 
+fn burn_transferred_assets<D, H>(
+    wl_storage: &mut WlStorage<D, H>,
+    transfer: &PendingTransfer,
+) -> Result<BTreeSet<Key>>
+where
+    D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
+    H: 'static + StorageHasher + Sync,
+{
+    let mut changed_keys = BTreeSet::default();
+
+    let native_erc20_addr = match wl_storage
```

Comment:
doesn't storage already have a method that returns an error if deserialization fails?